app/WebParsing/UtsgJsonParser.hs

```diff
@@ -0,0 +1,251 @@
+                                                                                               else (T.unpack $ fromJust $ firstNameSection !! 0) ++ " " ++ (T.unpack $ fromJust $ lastNameSection !! 0))
+                                                                          firstNameSections
+                                                                          lastNameSections)
+                                            allInstructorsFirstname
+                                            allInstructorsLastname
+
+    let times                     = zipWith3 (\courseDays
+                                               courseStarts
+                                               courseEnds -> zipWith3 (\dayLst
+                                                                        startLst
+                                                                        endLst -> getNestedTimeSlots dayLst startLst endLst)
+                                                                      courseDays
+                                                                      courseStarts
+                                                                      courseEnds)
+                                             meetingDays
+                                             meetingStartTimes
+                                             meetingEndTimes
+
+    let timeStr                   = zipWith3 (\courseDays
+                                               courseStarts
+                                               courseEnds -> zipWith3 (\dayLst
+                                                                        startLst
+                                                                        endLst -> zipWith3 (\days start end -> if isJust days
+                                                                                                               then (T.unpack $ fromJust days) ++ " " ++ (T.unpack $ fromJust start) ++ "-" ++ (T.unpack $ fromJust end)
+                                                                                                               else "TBA")
+                                                                                            dayLst
+                                                                                            startLst
+                                                                                            endLst)
+                                                                      courseDays
+                                                                      courseStarts
+                                                                      courseEnds)
+                                             meetingDays
+                                             meetingStartTimes
+                                             meetingEndTimes
+
+    let manualTutorialEnrolments  = repeat      $ Just False
```

Comment:
Compute these two *after* computing the lecture objects. The point is to search through the corresponding list of lecture objects, and see if there's a "tutorial" section and a "laboratory" section. You can do this using a `map`.